metals/src/main/scala/scala/meta/internal/implementation/ImplementationProvider.scala

```diff
@@ -338,17 +338,29 @@ final class ImplementationProvider(
       file: Path
   ): Map[Path, Set[ClassLocation]] = {
 
+    def isTypeAlias(symbol: String) =
+      classContext
+        .findSymbol(symbol)
+        .exists(_.kind == SymbolInformation.Kind.TYPE)
+
     def loop(symbol: String, currentPath: Option[Path]): Set[ClassLocation] = {
       val directImplementations = classContext.getLocations(symbol).filterNot {
         loc =>
-          // we are not interested in local symbols from outside the workspace
-          (loc.symbol.isLocal && loc.file.isEmpty) ||
-          // local symbols inheritance should only be picked up in the same file
-          (loc.symbol.isLocal && loc.file != currentPath)
+          // we keep type aliases
+          !isTypeAlias(loc.symbol) &&
+          (
+            // we are not interested in local symbols from outside the workspace
+            (loc.symbol.isLocal && loc.file.isEmpty) ||
+            // local symbols inheritance should only be picked up in the same file
+            (loc.symbol.isLocal && loc.file != currentPath)
+          )
       }
       directImplementations ++ directImplementations
         .flatMap { loc =>
-          val allPossible = loop(loc.symbol, loc.file)
+          val allPossible = loop(
+            loc.symbol,
+            loc.file.orElse(Some(file).filter(_ => isTypeAlias(loc.symbol)))
```

Comment:
It seems we are missing the file due to [here](https://github.com/scalameta/metals/blob/39fa3cc3acbed1b7e252747e520ce42ad542260f/metals/src/main/scala/scala/meta/internal/implementation/ImplementationProvider.scala#L490)

I think this was done in order not to show types as implementations. We could add the file the same way we add it in the method above and move checking for type alias in [here](https://github.com/scalameta/metals/blob/39fa3cc3acbed1b7e252747e520ce42ad542260f/metals/src/main/scala/scala/meta/internal/implementation/ImplementationProvider.scala#L323)

We should use the `implDocument` to search for the `SymbolInformation` and check if it is a type alias.